File: TiDB_shopping_frontend/src/types/order.ts
/**
 * Represents an item when creating an order.
 * Backend's OrderItemBase for creation requires product_id and quantity.
 */
export interface OrderItemForCreation {
  product_id: number;  // Changed to match backend expectation
  quantity: number;
}

/**
 * Payload for creating a new order.
 * This should match the fields expected by the POST /api/orders endpoint.
 */
export interface OrderCreationPayload {
  items: OrderItemForCreation[];
}

/**
 * Represents an item within a retrieved order.
 * Corresponds to backend's OrderItemOut with snake_case field names.
 */
export interface OrderItem {
  id: string;
  product_id: number;
  product_name: string;
  quantity: number;
  price: number; // Price per unit at the time of purchase
}

/**
 * Represents a customer's order as retrieved from the backend.
 * Corresponds to backend's OrderOut with snake_case field names.
 */
export interface Order {
  id: string;
  order_number: string;
  user_id: string;
  order_date: string; // Typically ISO string from backend
  total_amount: number;
  status: 'PENDING' | 'PROCESSING' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED' | string; // Allow string for future statuses
  items: OrderItem[];
}

// Response from creating an order is the newly created Order itself
export type OrderCreationResponse = Order;

File: TiDB_shopping_backend/schemas/order.py
from pydantic import BaseModel
from typing import List
from datetime import datetime
from schemas.order_item import OrderItemBase, OrderItemOut

class OrderBase(BaseModel):
    order_number: str
    total_amount: float
    status: str

class OrderOut(OrderBase):
    id: str
    order_date: datetime
    user_id: str
    items: List[OrderItemOut]
    class Config:
        orm_mode = True

class OrderCreationRequest(BaseModel):
    items: List[OrderItemBase]